verless serve runs into race condition when another instance already is serving

### Description

Serving a verless project works fine, until you try to serve another project while the first serve is still running:

```
2020/10/01 23:24:07 rebuild
panic: send on closed channel

goroutine 1 [running]:
github.com/verless/verless/core.RunServe(0xc00001e0d0, 0x7, 0x0, 0x0, 0x0, 0x1f90, 0xc0001cba70, 0x4, 0x4, 0x0, ...)
        /home/circleci/project/core/serve.go:125 +0x3d3
github.com/verless/verless/cli.newServeCmd.func1(0xc0002c6b00, 0xc000587fe0, 0x1, 0x1, 0x0, 0x0)
        /home/circleci/project/cli/serve.go:20 +0xbf
github.com/spf13/cobra.(*Command).execute(0xc0002c6b00, 0xc000587fc0, 0x1, 0x1, 0xc0002c6b00, 0xc000587fc0)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x45a
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002718c0, 0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /home/circleci/project/cmd/verless/main.go:12 +0x2e
```

If a project already is served on the same address, verless probably closes one or more channel and another goroutine tries to send from that channel because it doesn't know that serving is not possible.

### Reproduce

Steps to reproduce the behavior:
1. `verless serve example`
2. `verless serve example` in new terminal

### verless version

The output of `verless version`:

```
verless version v0.4.3
Git tag: v0.4.3
Git commit: b96616f
```

### System information

* OS: Windows 10
* 1909

--- a/core/serve.go
+++ b/core/serve.go
@@ -2,16 +2,14 @@ package core
 
 import (
 	"fmt"
-	"net"
-	"net/http"
-	"path/filepath"
-	"sync"
-
 	"github.com/spf13/afero"
 	"github.com/verless/verless/config"
 	"github.com/verless/verless/out"
 	"github.com/verless/verless/out/style"
 	"github.com/verless/verless/theme"
+	"net"
+	"net/http"
+	"path/filepath"
 )
 
 // ServeOptions represents options for running a verless listenAndServe command.
@@ -47,7 +45,6 @@ func Serve(path string, options ServeOptions) error {
 	done := make(chan bool)
 	rebuildCh := make(chan string)
 
-	// Only watch if needed.
 	if options.Watch {
 		if err := watch(watchContext{
 			IgnorePaths: []string{
@@ -63,63 +60,47 @@ func Serve(path string, options ServeOptions) error {
 		}
 	}
 
-	var initialBuild sync.WaitGroup
+	out.T(style.Sparkles, "building project ...")
 
-	// Start rebuild goroutine.
-	// If watch is not enabled, it's still used for the initial build.
-	go func() {
-		isFirst := true
-		initialBuild.Add(1)
+	build, err := NewBuild(memMapFs, path, options.BuildOptions)
+	if err != nil {
+		return err
+	}
 
-		for {
-			select {
-			case _, ok := <-rebuildCh:
-				finishFirst := func() {
-					if isFirst {
-						initialBuild.Done()
-						isFirst = false
+	if err := build.Run(); err != nil {
+		return err
+	}
+
+	out.T(style.HeavyCheckMark, "project built successfully")
+
+	// If --watch is enabled, launch a goroutine that handles rebuilds.
+	if options.Watch {
+		go func() {
+			for {
+				select {
+				case _, ok := <-rebuildCh:
+					if !ok {
+						return
 					}
-				}
+					out.T(style.Sparkles, "rebuilding project ...")
 
-				if !ok {
-					return
-				}
-				out.T(style.Sparkles, "building project ...")
+					build, err := NewBuild(memMapFs, path, options.BuildOptions)
+					if err != nil {
+						out.Err(style.Exclamation, "failed to initialize new build: %s", err.Error())
+						continue
+					}
 
-				build, err := NewBuild(memMapFs, path, options.BuildOptions)
-				if err != nil {
-					out.Err(style.Exclamation, "failed to initialize new build: %s", err.Error())
-					finishFirst()
-					continue
-				}
+					if err := build.Run(); err != nil {
+						out.Err(style.Exclamation, "failed to build the project: %s", err.Error())
+						continue
+					}
 
-				if err := build.Run(); err != nil {
-					out.Err(style.Exclamation, "failed to build the project: %s", err.Error())
-					finishFirst()
-					continue
-				}
-
-				out.T(style.HeavyCheckMark, "project built successfully")
-
-				finishFirst()
-			case _, ok := <-done:
-				// Stops the goroutine if requested to.
-				// Triggers on closing of the done channel.
-				if !ok {
+					out.T(style.HeavyCheckMark, "project built successfully")
+				case _, _ = <-done:
 					return
 				}
 			}
-		}
-	}()
-
-	// Trigger and wait for the initial rebuild.
-	rebuildCh <- path
-	initialBuild.Wait()
-
-	// Stop rebuilding goroutine if not watching.
-	if !options.Watch {
-		done <- true
-		close(done)
+		}()
 	}
 
 	// If the target folder doesn't exist, return an error.
@@ -128,9 +109,6 @@ func Serve(path string, options ServeOptions) error {
 	}
 
 	err = listenAndServe(memMapFs, targetFiles, options.IP, options.Port)
-
-	// Stop building goroutine just to be sure.
-	done <- true
 	close(done)
 
 	return err

--- a/core/serve.go
+++ b/core/serve.go
@@ -2,14 +2,15 @@ package core
 
 import (
 	"fmt"
+	"net"
+	"net/http"
+	"path/filepath"
+
 	"github.com/spf13/afero"
 	"github.com/verless/verless/config"
 	"github.com/verless/verless/out"
 	"github.com/verless/verless/out/style"
 	"github.com/verless/verless/theme"
-	"net"
-	"net/http"
-	"path/filepath"
 )
 
 // ServeOptions represents options for running a verless listenAndServe command.

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,6 @@ module github.com/verless/verless
 go 1.14
 
 require (
-	github.com/cweill/gotests v1.5.3 // indirect
 	github.com/fsnotify/fsnotify v1.4.9 // indirect
 	github.com/google/go-cmp v0.5.2
 	github.com/gorilla/feeds v1.1.1
@@ -20,6 +19,6 @@ require (
 	github.com/yuin/goldmark-highlighting v0.0.0-20200307114337-60d527fdb691
 	github.com/yuin/goldmark-meta v0.0.0-20191126180153-f0638e958b60
 	golang.org/x/sys v0.0.0-20200803210538-64077c9b5642 // indirect
-	golang.org/x/tools v0.0.0-20201010145503-6e5c6d77ddcc // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
 	gopkg.in/ini.v1 v1.57.0 // indirect
 )

